mailpoet/mailpoet
# New WooCommerce segment: single order value [MAILPOET-4988]

## Description

This PR adds a new WooCommerce segment called "single order value". It allows users to create segments based on the value of a single order made by customers. For example, customers that placed an order with a total value equal to or higher than $1000 in the last seven days.

## Code review notes

I based this segment on the WooCommere total spent segment.

## QA notes

To test this new segment:

- Create a few WooCommerce others with different customers. This can be done via the web interface or using [wc-smooth-generator](https://github.com/woocommerce/wc-smooth-generator).
- Go to `/wp-admin/admin.php?page=mailpoet-segments#/new-segment` and create a new segment of the type `single order value`.
- Send an email using this segment and confirm that the recipients of the email are the expected WooCommerce customers that placed at least one order that matched the criteria chosen when creating the segment.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4988]

## After-merge notes

_N/A_


[MAILPOET-4988]: https://mailpoet.atlassian.net/browse/MAILPOET-4988?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
